src/core/parser.zsh

```diff
@@ -168,3 +168,37 @@ _fzf_complete_parse_option_arguments() {
 
     echo - $result
 }
+
+_fzf_complete_parse_option() {
+    local result=()
+    local short=$1
+    local long=$2
+    shift 2
+
+    local shorts=${(pj::)${${(z)short}#*-}}
+    local longs=${(pj:|:)${(z)long}}
+
+    local cmd=(${(Q)${(z)@}})
+    local idx=1
+    while [[ $idx -le ${#cmd} ]]; do
+        if [[ -n $shorts ]]; then
+            if [[ ${cmd[idx]} =~ ^-([^\-$shorts]*[$shorts]+[^$shorts]*)+$ ]]; then
+                result+=(${(qq)cmd[idx]})
+            fi
+        fi
+
+        if [[ -n $longs ]]; then
+            if [[ ${cmd[idx]} =~ $longs ]]; then
```

Comment:
More exactly: `[[ ${cmd[idx]} = $longs ]]`, isn't it? I couldn't find any reason for using `=~`.

Reply:
`$longs` is separated by `|` as `--soft|--hard|--merge|--keep`. The `|` operator exists in the glob, but setting the glob to a variable isn't work.

Reply:
I don't like that design. Presumably $longs should be put into an array `(--soft --hard --merge --keep)` and simply referening it by `(r)` operator seems enough.

Reply:
`(r)` operator uses glob, too. There is the method of looping `(--soft --hard --merge --keep)` otherwise, though.

Reply:
I don't understand what you are talking about.